### DISCUSSION TITLE
Solving the two transport equations for the standard k epsilon turbulence model

### DISCUSSION THREAD

**Discussion by Leon-novice** [zwicker-group/py-pde#631] [Q&A]
I am relatively new to python and pde solvers in general. I was very pleased to come across py-pde and have had some success in solving the Spalart Allmaras turbulence equation using it. However the standard ke turbulence model has coefficients associated with the derivative operators and I am having difficulties in passing these into the pde equation. I have looked at the examples but could not find any seemed to be relevant. I have tried using 'scalarFields' to pass the coefficient arrays but without success. I would be very grateful for any guidance on how to pass data arrays into the pde equation. Thank you.

**Comment by david-zwicker** [comment]
Please provide details of the equations you want to implement, demonstrate what you have tried, what error you encounter, and what result you expect instead.

**Reply by Leon-novice** [reply]
![image](https://github.com/user-attachments/assets/d59af191-d67d-43da-94e2-7d19230053d5)

Hello David, the equations to be solved for 'k' and 'epsilon' are as above. 'ui' are velocity fields and 'Mut' and 'con' are scalar fields.. All fields have the same size My problem is that I don't know how to access these fields within the pde. 

![Screenshot 2024-11-30 132927](https://github.com/user-attachments/assets/92ca9dfe-894c-4174-b342-a043945c47f9)

The error that I get is 

![Screenshot 2024-11-30 133548](https://github.com/user-attachments/assets/5147c763-0409-4b6a-80e1-c2baff604805)
![Screenshot 2024-11-30 133608](https://github.com/user-attachments/assets/d73064c5-e8d0-44fe-9ff6-5a65e8af52bd)

As for results I would hope for fields of values for 'k'  between 0 and 100 and for 'epsilon' between 1000 and 2000. This is a steep learning curve for me and I realise that there will be other errors in the code and I haven't yet worked out how to specify the initial conditions but some assistance with the coefficients would be a great start.

Many thanks Leon

**Reply by david-zwicker** [reply]
If I understand you correctly, you want to use these fields as constants during the evolution of the PDEs. If this is the case, you could simply pass a dictionary of these fields as the `consts` argument of the `PDE` class; see [documentation](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.pde.html#pde.pdes.pde.PDE). You can then use the keys of this dictionary as named variables in the expression. A short example is

```python
eq = pde.PDE({"k": "laplace(k) / u"}, consts={"u": u})
```
where the field `u` is made available in the expression.

**Reply by Leon-novice** [reply]
Thank you David I think that is exactly what I want to do. I will give it a try.
